### PR TITLE
Nodepool inventory script

### PR DESCRIPTION
**Pull request by rcarrillocruz**
##### SUMMARY

This is an inventory plugin for the OpenStack Infra Nodepool service.
For more info, https://docs.openstack.org/infra/nodepool/feature/zuulv3/.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

contrib/inventory/nodepool.py

##### ANSIBLE VERSION

```
ansible 2.5.0 (nodepool_dyn_inv 0562486c7e) last updated 2018/01/28 18:30:14 (GMT +200)

```

### CODE REVIEW

File: contrib/inventory/nodepool.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            label = node['type']
+            state = node['state']
+            provider = node['provider']
+            image = node['image_id']
+            data['_meta']['hostvars'][n] = {'ansible_host': node['interface_ip'],
+                                            'ansible_port': node['connection_port'],
+                                            'ansible_connection': node['connection_type'],
+                                            'ansible_user': node['username'],
+                                            'nodepool_label': label,
+                                            'nodepool_state': state,
```
Comment (odyssey4me) [inline contrib/inventory/nodepool.py]: It seems that state is collected here, but not output. Is that intentional?

File: contrib/inventory/nodepool.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+Generates an inventory that Ansible can understand by making Zookeeper calls
+to the Nodepool service using the kazoo library.
+Zookeeper IP and port are read from environment variables NODEPOOL_ZK_IP and
+NODEPOOL_ZK_PORT respectively. If those variables are not set, the script defaults
+to 127.0.0.1 for the address and 2181 for the port.
+Example:
+
+  #!/bin/bash
+  export NODEPOOL_ZK_IP=10.10.10.10
+  export NODEPOOL_ZK_PORT=9000
```
Comment (odyssey4me) [inline contrib/inventory/nodepool.py:17]: It'd be super-nice if the standard nodepool yaml config file (/etc/nodepool/nodepool.yaml IIRC) could also be parsed by default, so if it's there, it gets used.

File: contrib/inventory/nodepool.py (hunk 1 above)
Comment (rcarrillocruz) [reply contrib/inventory/nodepool.py]: Not sure I follow, on my test nodepool I do get that output on metavars:

https://gist.github.com/rcarrillocruz/965fb1dd1887aac7e09d4424c73f3cda

File: contrib/inventory/nodepool.py (hunk 2 above)
Comment (rcarrillocruz) [reply contrib/inventory/nodepool.py:17]: Done

**Comment by maxamillion** [comment]
@rcarrillocruz Any chance this could be a dynamic inventory plugin instead of the old-style script?

**Pull request closed without merging**